File: mixed-persistence/mixed-persistence-scala-sbt/hello-api/src/main/scala/com/lightbend/lagom/samples/hello/api/HelloService.scala
package com.lightbend.lagom.samples.hello.api

import akka.{ Done, NotUsed }
import com.lightbend.lagom.scaladsl.api.{ Service, ServiceCall }
import play.api.libs.json.{ Format, Json }

trait HelloService extends Service {

  def hello(id: String): ServiceCall[NotUsed, String]

  def useGreeting(id: String): ServiceCall[GreetingMessage, Done]

  def allGreetings(): ServiceCall[NotUsed, Seq[Greeting]]

  override final def descriptor = {
    import Service._
    named("hello")
      .withCalls(
        pathCall("/api/hello/:id", hello _),
        pathCall("/api/hello/:id", useGreeting _),
        pathCall("/api/greetings", allGreetings)
      )
      .withAutoAcl(true)
  }
}

/**
  * The Greeting is both the message and the person that message is meant for.
  */
case class Greeting(name: String, message: String)

object Greeting {
  implicit val format: Format[Greeting] = Json.format
}

/**
  * This class only models the message of a Greeting
  */
case class GreetingMessage(message: String)

object GreetingMessage {
  implicit val format: Format[GreetingMessage] = Json.format[GreetingMessage]
}

File: lightbend-telemetry/lightbend-telemetry-java-mvn/hello-api/src/main/java/com/lightbend/lagom/samples/cinnamon/hello/api/HelloService.java
package com.lightbend.lagom.samples.cinnamon.hello.api;

import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.Service;
import com.lightbend.lagom.javadsl.api.ServiceCall;

import static com.lightbend.lagom.javadsl.api.Service.named;
import static com.lightbend.lagom.javadsl.api.Service.pathCall;

public interface HelloService extends Service {

    /**
     * Example: curl http://localhost:9000/api/hello/Alice
     */
    ServiceCall<NotUsed, String> hello(String id);


    /**
     * Example: curl http://localhost:9000/api/hello-proxy/Alice
     */
    ServiceCall<NotUsed, String> helloProxy(String id);

    @Override
    default Descriptor descriptor() {
        return named("hello").withCalls(
                pathCall("/api/hello/:id", this::hello),
                pathCall("/api/hello-proxy/:id", this::helloProxy)
        ).withAutoAcl(true);
    }

}
